lib/core/macros.nim

```diff
@@ -1626,9 +1626,18 @@ macro unpackVarargs*(callee: untyped; args: varargs[untyped]): untyped =
     result.add args[i]
 
 proc getProjectPath*(): string = discard
-  ## Returns the path to the currently compiling project, not to
-  ## be confused with ``system.currentSourcePath`` which returns
-  ## the path of the current module.
+  ## Returns the path to the currently compiling project,
+  ## the path of the Nim file being compiled,
+  ## not to be confused with ``system.currentSourcePath`` which returns
+  ## the path of the current module. Its available at compile-time.
+  ## Its available for JavaScript, NodeJS and NimScript backends.
+  ## Its available for C, C++ and ObjectiveC backends.
```

Comment:
I would just omit this entirely; if there is nothing special about a proc's behavior for a backend, it's probably safe to assume that it works with every backend.